recipes/libpqxx/all/conanfile.py

```diff
@@ -0,0 +1,105 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+import os
+
+
+class LibpqxxRecipe(ConanFile):
+    name = "libpqxx"
+    settings = "os", "compiler", "build_type", "arch"
+    description = "The official C++ client API for PostgreSQL"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jtv/libpqxx"
+    license = "BSD-3-Clause"
+    topics = ("conan", "libpqxx", "postgres", "postgresql", "database", "db")
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    requires = "libpq/11.5"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler = str(self.settings.compiler)
+        compiler_version = Version(self.settings.compiler.version.value)
+
+        minimal_version = {
+            "Visual Studio": "15",
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10"
+        }
+
+        if compiler in minimal_version and \
+           compiler_version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a compiler that supports"
+                                            " at least C++17. %s %s is not"
+                                            " supported." % (self.name, compiler, compiler_version))
+        minimal_cpp_standard = "17"
+        supported_cppstd = ["17", "20"]
+
+        if not self.settings.compiler.cppstd:
+            self.output.info("Settings c++ standard to {}".format(minimal_cpp_standard))
+            self.settings.compiler.cppstd = minimal_cpp_standard
+
+        if not self.settings.compiler.cppstd in supported_cppstd:
+            raise ConanInvalidConfiguration(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+
+        if self.options.shared and \
+           self.settings.os == "Windows" and \
+           compiler == "Visual Studio":
+            self.output.info("Override libpq:shared to True.")
+            self.options["libpq"].shared = True
```

Comment:
When libpqxx is a shared library and libpq is a static library, I got unresolved reference in My Machine and AppVeyor: https://ci.appveyor.com/project/bincrafters/conan-libpqxx/builds/30502288/job/vhsx9hylygojes73#L4670

I could not solve this problem.
However, AppVeyor has been successful several times.

I try removing the code once.

Reply:
I removed it but cannot build now.

The latest error below seems to be still different. Is it a problem of libpq?

<details>
<summary>build log</summary>
```bash
[HOOK - conan-center.py] pre_export(): [DEPRECATED GLOBAL CPPSTD (KB-H001)] OK
[HOOK - conan-center.py] pre_export(): [REFERENCE LOWERCASE (KB-H002)] OK
[HOOK - conan-center.py] pre_export(): [RECIPE METADATA (KB-H003)] OK
[HOOK - conan-center.py] pre_export(): [HEADER_ONLY, NO COPY SOURCE (KB-H005)] OK
[HOOK - conan-center.py] pre_export(): [FPIC OPTION (KB-H006)] OK
[HOOK - conan-center.py] pre_export(): [VERSION RANGES (KB-H008)] OK
[HOOK - conan-center.py] pre_export(): [RECIPE FOLDER SIZE (KB-H009)] Total recipe size: 5.6328125 KB
[HOOK - conan-center.py] pre_export(): [RECIPE FOLDER SIZE (KB-H009)] OK
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] exports: None
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] exports: ['CMakeLists.txt', 'patches/*']
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE FOLDER (KB-H024)] OK
[HOOK - conan-center.py] pre_export(): [META LINES (KB-H025)] OK
[HOOK - conan-center.py] pre_export(): [CONAN CENTER INDEX URL (KB-H027)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE MINIMUM VERSION (KB-H028)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE - RUN ENVIRONMENT (KB-H029)] OK
[HOOK - conan-center.py] pre_export(): [SYSTEM REQUIREMENTS (KB-H032)] OK
[HOOK - conan-center.py] pre_export(): [CONANDATA.YML FORMAT (KB-H030)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE - NO IMPORTS() (KB-H034)] OK
[HOOK - conan-center.py] pre_export(): [NO AUTHOR (KB-H037)] OK
[HOOK - conan-center.py] pre_export(): [NO TARGET NAME (KB-H040)] OK
Exporting package recipe
libpqxx/7.0.1 exports: File 'conandata.yml' found. Exporting it...
libpqxx/7.0.1 exports: Copied 1 '.yml' file: conandata.yml
libpqxx/7.0.1 exports_sources: Copied 1 '.txt' file: CMakeLists.txt
libpqxx/7.0.1 exports_sources: Copied 1 '.patch' file: 0001-cmake-fix-module.patch
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] Saving conandata.yml: {'sources': {'7.0.1': {'url': 'https://github.com/jtv/libpqxx/archive/7.0.1.tar.gz', 'sha256': '8c607ea4142223823ec400a3ff8f9561e8674e5888243cb7c6a610bf548ae0f0'}}, 'patches': {'7.0.1': [{'patch_file': 'patches/0001-cmake-fix-module.patch', 'base_path': 'source_subfolder'}]}}
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] New conandata.yml contents: patches:
  7.0.1:
  - base_path: source_subfolder
    patch_file: patches/0001-cmake-fix-module.patch
sources:
  7.0.1:
    sha256: 8c607ea4142223823ec400a3ff8f9561e8674e5888243cb7c6a610bf548ae0f0
    url: https://github.com/jtv/libpqxx/archive/7.0.1.tar.gz

[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] OK
libpqxx/7.0.1: A new conanfile.py version was exported
libpqxx/7.0.1: Folder: G:\.conan\data\libpqxx\7.0.1\_\_\export
libpqxx/7.0.1: Package recipe modified in export, forcing source folder removal
libpqxx/7.0.1: Use the --keep-source, -k option to skip it
libpqxx/7.0.1: Exported revision: 9249c4881c2edb6ec85cc41ba5dfb1cf
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.cppstd=17
compiler.runtime=MD
compiler.version=15
os=Windows
os_build=Windows
[options]
libpqxx:shared=True
[build_requires]
[env]

libpqxx/7.0.1: Forced build from source
libpqxx/7.0.1 (test package): Installing package
Requirements
    libpq/11.5 from 'conan-center' - Cache
    libpqxx/7.0.1 from local cache - Cache
    zlib/1.2.11 from 'conan-center' - Cache
Packages
    libpq/11.5:66397d4e4f72ab2f3a4c316d3e5132059c9ebf7b - Cache
    libpqxx/7.0.1:f8890a8c97b806406ae147dbcc7184e8742b1627 - Build
    zlib/1.2.11:6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7 - Cache

zlib/1.2.11: Already installed!
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
libpq/11.5: Already installed!
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] pre_source(): [IMMUTABLE SOURCES (KB-H010)] OK
libpqxx/7.0.1: Configuring sources in G:\.conan\data\libpqxx\7.0.1\_\_\source

[HOOK - conan-center.py] post_source(): [LIBCXX MANAGEMENT (KB-H011)] OK
[HOOK - conan-center.py] post_source(): [CPPSTD MANAGEMENT (KB-H022)] OK
libpqxx/7.0.1: Copying sources to build folder
libpqxx/7.0.1: Building your package in G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627
libpqxx/7.0.1: Generator cmake created conanbuildinfo.cmake
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] 'fPIC' option not found
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
libpqxx/7.0.1: Calling build()
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.18362.
-- The C compiler identification is MSVC 19.16.27034.0
-- The CXX compiler identification is MSVC 19.16.27034.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan setting CPP STANDARD: 17 WITH EXTENSIONS OFF
-- Found PostgreSQL: G:/.conan/data/libpq/11.5/_/_/package/66397d4e4f72ab2f3a4c316d3e5132059c9ebf7b/lib/libpq.lib (found version "11.5") 
-- Looking for poll
-- Looking for poll - not found
-- Looking for PQencryptPasswordConn
-- Looking for PQencryptPasswordConn - not found
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Generating config.h
-- Generating config.h - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY


-- Build files have been written to: G:/.conan/data/libpqxx/7.0.1/_/_/build/f8890a8c97b806406ae147dbcc7184e8742b1627/build_subfolder
.NET Framework  Microsoft (R) Build Engine o[W 15.9.21+g9802d43bc3

Copyright (C) Microsoft Corporation.All rights reserved.



  Checking Build System

  Building Custom Rule G:/.conan/data/libpqxx/7.0.1/_/_/build/f8890a8c97b806406ae147dbcc7184e8742b1627/source_subfolder/src/CMakeLists.txt

  array.cxx

  binarystring.cxx

  connection.cxx

  cursor.cxx

  encodings.cxx

  errorhandler.cxx

  except.cxx

  field.cxx

  largeobject.cxx

  notification.cxx

  pipeline.cxx

  result.cxx

  robusttransaction.cxx

  row.cxx

  sql_cursor.cxx

  statement_parameters.cxx

  strconv.cxx

  stream_from.cxx

  stream_to.cxx

  subtransaction.cxx

  transaction.cxx

  transaction_base.cxx

  util.cxx

  version.cxx

     Cu G:/.conan/data/libpqxx/7.0.1/_/_/build/f8890a8c97b806406ae147dbcc7184e8742b1627/build_subfolder/lib/pqxx.lib IuWFNg G:/.conan/data/libpqxx/7.0.1/_/_/build/f8890a8c97b806406ae147dbcc7184e8742b1627/build_subfolder/lib/pqxx.exp 

libpq.lib(fe-exec.obj) : error LNK2019: OV{ pg_vsnprintf  pqInternalNotice QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(pqexpbuffer.obj) : error LNK2001: OV{ "pg_vsnprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(chklocale.obj) : error LNK2001: OV{ "pg_sprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(inet_net_ntop.obj) : error LNK2001: OV{ "pg_sprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(win32.obj) : error LNK2001: OV{ "pg_sprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(scram-common.obj) : error LNK2001: OV{ "pg_sprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-exec.obj) : error LNK2001: OV{ "pg_sprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2001: OV{ "pg_sprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-auth.obj) : error LNK2001: OV{ "pg_sprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-protocol2.obj) : error LNK2001: OV{ "pg_sprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(chklocale.obj) : error LNK2001: OV{ "pg_fprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-exec.obj) : error LNK2001: OV{ "pg_fprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2001: OV{ "pg_fprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(encnames.obj) : error LNK2001: OV{ "pg_fprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-misc.obj) : error LNK2001: OV{ "pg_fprintf" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ pg_snprintf  PQconnectPoll QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(thread.obj) : error LNK2001: OV{ "strlcpy" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2001: OV{ "strlcpy" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-protocol2.obj) : error LNK2001: OV{ "strlcpy" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-protocol3.obj) : error LNK2001: OV{ "strlcpy" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(ip.obj) : error LNK2001: OV{ "strlcpy" B [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ pg_gai_strerror  PQconnectPoll QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_FreeCredentialsHandle  pqDropConnection QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_DeleteSecurityContext  pqDropConnection QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_connect  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_init  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_unbind  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_search_st  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_msgfree  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_err2string  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_first_entry  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_count_entries  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_get_values_len  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-connect.obj) : error LNK2019: OV{ __imp_ldap_value_free_len  ldapServiceLookup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-auth.obj) : error LNK2019: OV{ AcquireCredentialsHandleA  pg_SSPI_startup QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-auth.obj) : error LNK2019: OV{ InitializeSecurityContextA  pg_SSPI_continue QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(fe-auth.obj) : error LNK2019: OV{ FreeContextBuffer  pg_SSPI_continue QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(chklocale.obj) : error LNK2019: OV{ pgwin32_setlocale  pg_get_encoding_from_locale QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(ip.obj) : error LNK2019: OV{ pg_getaddrinfo  pg_getaddrinfo_all QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(ip.obj) : error LNK2019: OV{ pg_freeaddrinfo  pg_freeaddrinfo_all QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

libpq.lib(ip.obj) : error LNK2019: OV{ pg_getnameinfo  pg_getnameinfo_all QB [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\bin\pqxx.dll : fatal error LNK1120: 25 OQ [G:\.conan\data\libpqxx\7.0.1\_\_\build\f8890a8c97b806406ae147dbcc7184e8742b1627\build_subfolder\source_subfolder\src\pqxx.vcxproj]

````
</details>

Reply:
I built the libpq recipe locally, thinking it was a [this runtime library problem](https://github.com/microsoft/vcpkg/issues/9150), but could not resolve the link error.

Changing the link order between libpgport and libpq did not solve the problem.

Reply:
@tt4g This error is caused because the cmake build script of libpqxx improperly searches and sets the required linker flags for libpq. It assumes the only library needed is `libpq`.
This is mostly valid for a shared library, where the library knows of all the dependent libraries.

This approach fails for a static libpq. libpqxx will only link to `libpq.a` but not to any extra libraries which `libpq` needs.
Fixes for this can range from adding `link_libraries(${CONAN_LIBS})` at the root `CMakeLists.txt` file or patching the `libpqxx` cmake scripts to link to `CONAN_PKG::libpq` instead of `${LIBPQ_LIBRARIES}` or something similar (I did not check it myself).

Reply:
I checked out your pr, and I think there are 2 problems:
- the cmake script uses `FindPostgreSQL`, which is cmake's default: https://cmake.org/cmake/help/v3.14/module/FindPostgreSQL.html.
This default will assume that PostgreSQL is a shared library and not consider any dependencies, such as windows libraries.
- `libpq` has no `cmake_find_package` name set.

The fixes here are:
- fix `libpq` to add:
```
        self.cpp_info.names["cmake_find_package"] = "PostgreSQL"
        self.cpp_info.names["cmake_find_package_multi"] = "PostgreSQL"
```
- and add the `cmake_find_package` generator to your `libpqxx` recipe

Reply:
See https://github.com/conan-io/conan-center-index/pull/910 and https://github.com/tt4g/conan-center-index/pull/1